_patches/int8_linear_kernel_mps.py

```
@@ -340,17 +351,141 @@ def _try_int8_kernel_forward(self, input):
+            try:
+                import psutil
+                _DEQUANT_MODE = psutil.virtual_memory().total >= 48 * (1 << 30)
+                if not _DEQUANT_MODE:
+                    print(f"{TAG} ASFP8_INT8_DEQUANT=1 ignored: needs >= 48 GiB RAM "
+                          f"for the dequantised weight copy.")
+            except Exception:
+                _DEQUANT_MODE = False
+    return _DEQUANT_MODE
+
+
+def _maybe_dequant_weight(self, input):
```

Ruff (0.16.2)

[warning] Too many return statements (9 > 6)

(PLR0911)